### PR TITLE
release: v1.6.0

### PR DESCRIPTION
**Pull request by claytonparnell**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Staleness Report: 1.6.0(gpu)

Package | Current Version in the Distribution image | Latest Relevant Version in Upstream
---|---|---
numpy|1.26.4|1.26.4
jinja2|3.1.3|3.1.3
${\color{red}pandas}$|2.1.4|2.2.1
altair|5.2.0|5.2.0
${\color{red}boto3}$|1.34.51|1.34.60
ipython|8.22.2|8.22.2
jupyter-lsp|2.2.4|2.2.4
jupyterlab|4.1.4|4.1.4
amazon-codewhisperer-jupyterlab-ext|2.0.1|2.0.1
jupyter-scheduler|2.5.1|2.5.1
amazon-sagemaker-jupyter-scheduler|3.0.7|3.0.7
amazon-sagemaker-sql-magic|0.1.0|0.1.0
${\color{red}jupyterlab-lsp}$|5.0.3|5.1.0
amazon_sagemaker_sql_editor|0.1.3|0.1.3
${\color{red}scipy}$|1.11.4|1.12.0
scikit-learn|1.4.1.post1|1.4.1.post1
pip|23.3.2|23.3.2
${\color{red}torchvision}$|0.15.2|0.16.1
autogluon|0.8.2|0.8.2
ipywidgets|8.1.2|8.1.2
notebook|7.1.1|7.1.1
aws-glue-sessions|1.0.4|1.0.4
conda|23.11.0|23.11.0
${\color{red}fastapi}$|0.103.2|0.110.0
${\color{red}langchain}$|0.1.9|0.1.11
jupyter-ai|2.11.0|2.11.0
jupyter-dash|0.4.2|0.4.2
jupyter-server-proxy|4.1.0|4.1.0
jupyterlab-git|0.50.0|0.50.0
${\color{red}keras}$|2.12.0|2.15.0
matplotlib|3.8.3|3.8.3
nodejs|18.19.0|18.19.0
py-xgboost-gpu|1.7.6|1.7.6
thrift_sasl|0.4.3|0.4.3
pyhive|0.7.0|0.7.0
python-gssapi|1.8.3|1.8.3
python-lsp-server|1.10.0|1.10.0
${\color{red}pytorch-gpu}$|2.0.0|2.1.2
sagemaker-headless-execution-driver|0.0.12|0.0.12
sagemaker-jupyterlab-emr-extension|0.1.9|0.1.9
sagemaker-jupyterlab-extension|0.2.0|0.2.0
sagemaker-kernel-wrapper|0.0.2|0.0.2
sagemaker-python-sdk|2.212.0|2.212.0
sagemaker-studio-analytics-extension|0.0.21|0.0.21
sasl|0.3.1|0.3.1
supervisor|4.2.5|4.2.5
${\color{red}tensorflow}$|2.12.1|2.15.0
uvicorn|0.28.0|0.28.0

# Staleness Report: 1.6.0(cpu)

Package | Current Version in the Distribution image | Latest Relevant Version in Upstream
---|---|---
numpy|1.26.4|1.26.4
jinja2|3.1.3|3.1.3
${\color{red}pytorch}$|2.0.0|2.1.2
${\color{red}pandas}$|2.1.4|2.2.1
altair|5.2.0|5.2.0
${\color{red}boto3}$|1.34.51|1.34.60
ipython|8.22.2|8.22.2
jupyter-lsp|2.2.4|2.2.4
jupyterlab|4.1.4|4.1.4
amazon-codewhisperer-jupyterlab-ext|2.0.1|2.0.1
jupyter-scheduler|2.5.1|2.5.1
amazon-sagemaker-jupyter-scheduler|3.0.7|3.0.7
amazon-sagemaker-sql-magic|0.1.0|0.1.0
${\color{red}jupyterlab-lsp}$|5.0.3|5.1.0
amazon_sagemaker_sql_editor|0.1.3|0.1.3
${\color{red}scipy}$|1.11.4|1.12.0
scikit-learn|1.4.1.post1|1.4.1.post1
pip|23.3.2|23.3.2
${\color{red}torchvision}$|0.15.2|0.16.1
autogluon|0.8.2|0.8.2
ipywidgets|8.1.2|8.1.2
notebook|7.1.1|7.1.1
aws-glue-sessions|1.0.4|1.0.4
conda|23.11.0|23.11.0
${\color{red}fastapi}$|0.103.2|0.110.0
${\color{red}langchain}$|0.1.9|0.1.11
jupyter-ai|2.11.0|2.11.0
jupyter-dash|0.4.2|0.4.2
jupyter-server-proxy|4.1.0|4.1.0
jupyterlab-git|0.50.0|0.50.0
${\color{red}keras}$|2.12.0|2.15.0
matplotlib|3.8.3|3.8.3
nodejs|18.19.0|18.19.0
py-xgboost-cpu|1.7.6|1.7.6
thrift_sasl|0.4.3|0.4.3
pyhive|0.7.0|0.7.0
python-gssapi|1.8.3|1.8.3
python-lsp-server|1.10.0|1.10.0
sagemaker-headless-execution-driver|0.0.12|0.0.12
sagemaker-jupyterlab-emr-extension|0.1.9|0.1.9
sagemaker-jupyterlab-extension|0.2.0|0.2.0
sagemaker-kernel-wrapper|0.0.2|0.0.2
sagemaker-python-sdk|2.212.0|2.212.0
sagemaker-studio-analytics-extension|0.0.21|0.0.21
sasl|0.3.1|0.3.1
supervisor|4.2.5|4.2.5
${\color{red}tensorflow}$|2.12.1|2.15.0
uvicorn|0.28.0|0.28.0

### CODE REVIEW

**Comment by claytonparnell** [comment]
# WIP
## Stale Packages: 
| package      | reason for pin |
| ----------- | ----------- |
| pandas      |  autogluon.core requires >=1.4.1,<2.2 |
| boto3   | aiobotocore requires botocore>=1.34.41,<1.34.52        |
|jupyterlab-lsp|amazon_sagemaker_sql_editor (new pkg) requires jupyterlab-lsp==5.0.*|
|scipy|autogluon.tabular requires scipy >=1.5.4,<1.12|
|torchvision||
|fastapi||
|langchain||
|keras||
|tensorflow||

**Comment by claytonparnell** [comment]
=========================================================================================================================== short test summary info ==================================================================
==========================================================
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_gpu[keras.test.Dockerfile-required_packages0] - assert 1 == 0
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_gpu[scipy.test.Dockerfile-required_packages4] - assert 1 == 0
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_gpu[sm-python-sdk.test.Dockerfile-required_packages8] - assert 1 == 0
FAILED test/test_dockerfile_based_harness.py::test_dockerfiles_for_gpu[pandas.test.Dockerfile-required_packages7] - assert 1 == 0
====================================================================================================== 4 failed, 19 passed, 2 skipped, 16 warnings in 3797.35s (1:03:17) =============================================
==========================================================

NOTE - keras passed run individually
scipy and pandas failed with known failures tracked in #110 and #30

**Review by dylanraws — APPROVED** [review]
